chain/chunks/README.md

```diff
@@ -0,0 +1,30 @@
+# near-chunks
+
+This crate cotains functions to handle chunks. In NEAR - the block consists of multiple chunks - at most one per shard.
+
+When chunk is created, the creator encodes it using Reed Solomon encoding (ErasureCoding) - creating PartialChunks, that are later sent to all the validators (each validator gets a subset of them).
+
+Each validator is waiting to receive its share of PartialChunks (`has_all_parts`) - and once they are verified, it can accept the chunk (and if all chunks are accepted chain/client can sign the block).
+
+If validator cares about the given shard (that is - it wants to have a whole content of the shard present in the local memory) - it will want to receive the necessary amount of PartialChunks to be able to reconstruct the whole chunk. As we use ReedSolomon, this means that they need `data_shard_count` PartialChunks (which is lower than `total_shard_count`). Afterwards, it will reconstruct the chunk and persist it in the local cache (ChunkCache).
+
+
+## ShardsManager
+Shard Manager is responsible for:
+
+* **fetching partial chunks** - it can ask other nodes for the partial chunks that it is missing. It keeps the track fo the request via RequestPool
+* **handling partial chunks requests** - when request for the partial chunk arrives, it handles reading it from store and returning to the requesting node
+
+* **storing partial chunks** - it stores partial chunks within local cache (`stored_partial_encoded_chunks`)
```

Comment:
`stored_partial_encoded_chunks` is a temporary cache to store `PartialEncodedChunk`s before they can be added  to `ChunkCache`. With the existing implementation, we only merge a `PartialEncodedChunk` message in `ChunkCache` after it is verified, which requires the previous block of this chunk is processed. If the previous blocks are not processed, these `PartialEncodedChunk` messages are temporarily stored in `stored_partial_encoded_chunks`.

Also, not really affecting this PR, just fyi I plan to get rid of `stored_partial_encoded_chunks`

I would modify this section to be 
```**storing partial chunks** - it stores partial chunks within local cache before they can be used to reconstruct for the full chunk. `stored_partial_encoded_chunks` stores non-validated partial chunks while `ChunkCache` stores validated partial chunks```